src/Core/Settings.cpp

```diff
@@ -567,20 +567,20 @@ namespace ErrorCodes
     M(Bool, reject_expensive_hyperscan_regexps, true, "Reject patterns which will likely be expensive to evaluate with hyperscan (due to NFA state explosion)", 0) \
     M(Bool, allow_simdjson, true, "Allow using simdjson library in 'JSON*' functions if AVX2 instructions are available. If disabled rapidjson will be used.", 0) \
     M(Bool, allow_introspection_functions, false, "Allow functions for introspection of ELF and DWARF for query profiling. These functions are slow and may impose security considerations.", 0) \
-    M(Bool, splitby_max_substrings_includes_remaining_string, false, "Functions 'splitBy*()' with 'max_substrings' argument > 0 include the remaining string as last element in the result", 0) \
+    M(Bool, splitby_max_substrings_includes_remaining_string, false, "Functions 'splitBy*()' with 'max_substrings' argument > 0 include the remaining string as the last element in the result", 0) \
     \
     M(Bool, allow_execute_multiif_columnar, true, "Allow execute multiIf function columnar", 0) \
     M(Bool, formatdatetime_f_prints_single_zero, false, "Formatter '%f' in function 'formatDateTime()' prints a single zero instead of six zeros if the formatted value has no fractional seconds.", 0) \
     M(Bool, formatdatetime_parsedatetime_m_is_month_name, true, "Formatter '%M' in functions 'formatDateTime()' and 'parseDateTime()' print/parse the month name instead of minutes.", 0) \
     M(Bool, parsedatetime_parse_without_leading_zeros, true, "Formatters '%c', '%l' and '%k' in function 'parseDateTime()' parse months and hours without leading zeros.", 0) \
     M(Bool, formatdatetime_format_without_leading_zeros, false, "Formatters '%c', '%l' and '%k' in function 'formatDateTime()' print months and hours without leading zeros.", 0) \
     \
-    M(UInt64, max_partitions_per_insert_block, 100, "Limit maximum number of partitions in single INSERTed block. Zero means unlimited. Throw exception if the block contains too many partitions. This setting is a safety threshold, because using large number of partitions is a common misconception.", 0) \
+    M(UInt64, max_partitions_per_insert_block, 100, "Limit maximum number of partitions in the single INSERTed block. Zero means unlimited. Throw an exception if the block contains too many partitions. This setting is a safety threshold because using a large number of partitions is a common misconception.", 0) \
```

Comment:
Thanks for adding `the` / `a`. In this particular case, I am not sure if it makes the grammar better or not :-)

Maybe as a follow up: Most setting descriptions are very specific and require background/implementation knowledge. What would be extremely helpful is if we go through each description, check what it really does, what its side-effect are, and try to rewrite the description in a way that a random database DBA without much prior knowledge of ClickHouse internals could make sense out of it.

Reply:
Thank you for the feedback! I’m still learning about ClickHouse internals, so there might be a few things I need to adjust.

But in this particular case, I added "the" to refer to a specific inserted block and "a" to indicate any large number of partitions. I might be wrong, but this is how I understood the description.

Reply:
> Thanks for adding `the` / `a`. In this particular case, I am not sure if it makes the grammar better or not :-)
> 
> Maybe as a follow up: Most setting descriptions are very specific and require background/implementation knowledge. What would be extremely helpful is if we go through each description, check what it really does, what its side-effect are, and try to rewrite the description in a way that a random database DBA without much prior knowledge of ClickHouse internals could make sense out of it.



Thank you for the feedback! I’m still learning about ClickHouse internals, so there might be a few things I need to adjust.

But in this particular case, I added "the" to refer to a specific inserted block and "a" to indicate any large number of partitions. I might be wrong, but this is how I understood the description.